packages/yoroi-extension/app/api/ada/lib/state-fetch/remoteFetcher.js

```diff
@@ -383,29 +383,33 @@ export class RemoteFetcher implements IFetcher {
         method: 'get',
         timeout: 2 * CONFIG.app.walletRefreshInterval,
       }
-    ).then(response => response.data)
+    ).then(response => ({ error: null, data: response.data }))
       .catch((error) => {
-        if (error.response.status) {
-          Logger.info(`${nameof(RemoteFetcher)}::${nameof(this.getTokenInfo)} 404: no token meta found for subject: ` + id);
-        } else {
-          Logger.error(`${nameof(RemoteFetcher)}::${nameof(this.getTokenInfo)} error: ` + stringifyError(error));
+        if (error.response?.status === 404) {
+          return { error: 'noMetadata', data: id };
         }
-        return null;
+        Logger.error(`${nameof(RemoteFetcher)}::${nameof(this.getTokenInfo)} error: ` + stringifyError(error));
+        return { error: 'fail', data: null };
       }));
+    // return the mapping from query id/subject to token info
+    // if there is no info about a token (not an error), the value is null
+    // if there is an error querying a token, the key is not present
```

Comment:
So that the caller can distinguish "query successful but token not registered" from "query failure".